api/src/main/java/org/openmrs/validator/UserValidator.java

```diff
@@ -131,6 +133,13 @@ public void validate(Object obj, Errors errors) {
 					errors.rejectValue("email", "error.email.invalid");
 				}
 			}
+			
+			if (!StringUtils.isBlank(user.getUserActivationKey())) {
+				if(!isUserActivationKeyValid(user.getUserActivationKey())) {
```

Comment:
the activation key is system generated, why do you have to validate it against a certain pattern?